float8_experimental/float8_utils.py

```diff
@@ -34,17 +34,22 @@
 
 @torch.no_grad()
 def amax_to_scale(
-    amax: torch.Tensor, float8_dtype: torch.dtype, orig_dtype: torch.dtype
+    amax: torch.Tensor,
+    float8_dtype: torch.dtype,
+    orig_dtype: torch.dtype,
+    clamp_amax: bool = True,
 ):
     """Converts the amax value of a tensor to the fp8 scale.
     Args:
         amax: The amax value of the tensor.
         float8_dtype: The float8 dtype.
         orig_dtype: The original dtype of the tensor.
+        clamp_amax: default is True. False for FSDP fp8 all-gather since FSDP applied `torch.clamp` during pre-compute after optimizer.step
```

Comment:
good suggestion! I changed the API to precompute scale and it shows another 9% speed up *in unit test* vs precomputing amax

`fsdp_pre_all_gather` is also greatly simplified because of using `self._precomputed_scale`